### DISCUSSION TITLE
Error: Cannot find module 'redis' in service container

### DISCUSSION THREAD

**Discussion by amirensit** [community/community#26218] [Actions]
<p>I get the error: <code>Error: Cannot find module 'redis'</code> following this [tutorial](http:// <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/creating-redis-service-containers" class="inline-onebox-loading" rel="nofollow noopener">https://help.github.com/en/actions/configuring-and-managing-workflows/creating-redis-service-containers</a>) about connecting to redis service container.<br>
I even tried with:</p>
<pre><code>...

- name: Connect to Redis
        # Runs a script that creates a Redis client, populates
        # the client with data, and retrieves data
        run: |
          /usr/bin/docker exec -it redis  node ./.github/scripts/client.js
        # Environment variable used by the `client.js` script to create a new Redis client.
        env:
          # The hostname used to communicate with the Redis service container
          REDIS_HOST: redis
          # The default Redis port
          REDIS_PORT: 6379
</code></pre>
<p>but I get docker not found error.<br>
Any idea please ?</p>

**Comment by logankilpatrick** [comment]
<p>Hey! Just to be clear, this tutorial should work out of the box: <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/creating-redis-service-containers" rel="nofollow noopener">https://help.github.com/en/actions/configuring-and-managing-workflows/creating-redis-service-containers </a></p>
<p>If that is not the case, please contact GitHub support so they can fix the example. I suggest starting with that example workflow and then building your custom stuff in.</p>
<p>I also suggest checking out: <a href="https://github.com/marketplace?utf8=%E2%9C%93&amp;type=actions&amp;query=redis" rel="nofollow noopener">https://github.com/marketplace?utf8=✓&amp;type=actions&amp;query=redis</a> One of these may provide a way for you to simplify your workflow.</p>

**Comment by amirensit** [comment, marked as answer]
<p><a class="mention" href="/logankilpatrick">@logankilpatrick</a>  thanks for your reply.<br>
I find an example <a href="https://github.com/actions/example-services/tree/master/.github/workflows" rel="nofollow noopener">here</a> showing how to solve it.</p>
